### PR TITLE
Prl 6254

### PR DESCRIPTION
**Pull request by Scralfie**
### Change description

Event - C100MIAM

### JIRA link

[PRL-6254](https://tools.hmcts.net/jira/browse/PRL-6254)

**Before merging a pull request make sure that:**

- [x] Commits are meaningful and simple
- [x] All commits are squashed into a single commit
- [x] README and other documentation has been updated / added (if needed)

### CODE REVIEW

**Review by OLT2000 — DISMISSED** [review]
lgtm

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+export enum MiamPolicyUpgradeSubmitContent {
+  pageTitle = "MIAM",
+  h2 = "Check your answers",
+  text16CheckInfo = "Check the information below carefully.",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:5]: Whitespace

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  text16ChildrenInvolvedInEmergencyProtection = "*Are the children involved in any emergency protection, care or supervision proceedings (or have they been)?",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:7]: Whitespace

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  text16Attended1 = "*Are the children involved in any emergency protection, care or supervision proceedings (or have they been)?",
+  text16Attended2 = "*Has the applicant attended a Mediation Information & Assessment Meeting (MIAM)?",
+  text16Attended3 = "*Mediator Registration Number (URN)",
+  text16Attended4 = "*Family Mediation Service Name",
+  text16Attended5 = "*Sole Trader Name",
+  text16Attended6 = "*MIAM certificate",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:14]: Whitespace

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  text16CoreExemption1 = "*Are the children involved in any emergency protection, care or supervision proceedings (or have they been)?",
+  text16CoreExemption2 = "*Has the applicant attended a Mediation Information & Assessment Meeting (MIAM)?",
+  text16CoreExemption3 = "*Is the applicant claiming exemption from the requirement to attend a MIAM ?",
+  text16CoreExemption4 = "Why is the applicant not attending a MIAM?",
+  text16CoreExemption5 = "*What evidence of domestic abuse does the applicant have?",
+  text16CoreExemption6 = "*Can you provide evidence?",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:21]: Whitespace

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  text16Exemption1 = "Evidence",
+  text16Exemption2 = "*The applicant confirms that a child who would be the subject of the application or another child of the family who is living with that child is currently",
+  text16Exemption3 = "Why must the application be made urgently?",
+  text16Exemption4 = "Has, there been previous attendance of a MIAM or non-court dispute resolution?",
+  text16Exemption5 = "* Upload the MIAM certificate or evidence of participating in non-court dispute resolution",
+  text16Exemption6 = "What other grounds of exemption apply?",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:28]: Whitespace

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  text161Exemption1 = "Evidence",
+  text161Exemption2 = "*The applicant confirms that a child who would be the subject of the application or another child of the family who is living with that child is currently",
+  text161Exemption3 = "Why must the application be made urgently?",
+  text161Exemption4 = "Has, there been previous attendance of a MIAM or non-court dispute resolution?",
+  text161Exemption5 = "*What evidence of MIAM attendance are you submitting?",
+  text161Exemption6 = "*Upload the MIAM certificate signed by the mediator.",
+  text161Exemption7 = "What other grounds of exemption apply?",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:36]: Whitespace

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  text162Exemption1 = "Tell us why you cannot provide evidence",
+  text162Exemption2 = "*The applicant confirms that a child who would be the subject of the application or another child of the family who is living with that child is currently",
+  text162Exemption3 = "Why must the application be made urgently?",
+  text162Exemption4 = "Has, there been previous attendance of a MIAM or non-court dispute resolution?",
+  text162Exemption5 = "*What evidence of MIAM attendance are you submitting?",
+  text162Exemption6 = "*Provide the mediator details",
+  text162Exemption7 = "What other grounds of exemption apply?",
+  text162Exemption8 = "*Explain why the applicant is unable to attend MIAM",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:45]: Whitespace

File: e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  text16Yes = "Yes",
+  text16No = "No",
+  text16Change = "Change",
+
```
Comment (JakeCG) [inline e2e/fixtures/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitContent.ts:49]: Whitespace

File: e2e/pages/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgrade3Page.ts [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+      `${Selectors.GovukErrorMessage}:text-is("${MiamPolicyUpgrade3Content.errorMessageFileUpload}")`,
+      1,
+    );
+  }
+
+  private static async fillInFields({
+    page: page,
+    yesNoMiamPolicyUpgrade: yesNoMiamPolicyUpgrade,
+  }: fillinFieldsOptions): Promise<void> {
+    for (let i = 1; i <= 22; i++) {
```
Comment (JakeCG) [inline e2e/pages/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgrade3Page.ts:144]: This is fine for now, but in future you would have a radio enum that is separate for this, and you would do a for loop thats 
`(for radio in radioButtons) {`

File: e2e/pages/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitPage.ts [hunk 10]
```diff
@@ -0,0 +1,391 @@
...
+
+    switch (C100MiamPolicyUpgrade1PageType) {
+      case "yes":
+        await Helpers.checkVisibleAndPresent(
+          page,
+          `${Selectors.GovukText16}:text-is("${MiamPolicyUpgradeSubmitContent.text16ChildrenInvolvedInEmergencyProtection}")`,
+          1,
+        );
+        break;
+
```
Comment (JakeCG) [inline e2e/pages/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgradeSubmitPage.ts:118]: Whitespace

File: e2e/pages/manageCases/createCase/initialJourney/solicitorCreate4Page.ts [hunk 11]
```diff
@@ -20,7 +20,6 @@ export class SolicitorCreate4Page {
     }
     return await this.fillInFields(page);
   }
-
```
Comment (JakeCG) [inline e2e/pages/manageCases/createCase/initialJourney/solicitorCreate4Page.ts:23]: This should actually have Whitespace

File: e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+      user: "solicitor",
+      accessibilityTest: false,
+      errorMessaging: false,
+      C100MiamPolicyUpgrade1PageType: "yes",
+      yesNoMiamPolicyUpgrade: true,
+      miamSelection: "attended4MonthsPrior",
+      subJourney: true,
+    });
+  });
+  test(`Complete the C100 Create case MIAM as a solicitor with the following options:
```
Comment (JakeCG) [inline e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts:26]: Needs a gap

File: e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+  Not Accessibility testing,
+  Error message testing,
+  Saying yes to attended MIAM,
+  MiamSelection is attended4MonthsPrior. @crossbrowserManageCases`, async ({
```
Comment (JakeCG) [inline e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts:30]: Remove the tag here

File: e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+      user: "solicitor",
+      accessibilityTest: false,
+      errorMessaging: true,
+      C100MiamPolicyUpgrade1PageType: "yesAttendedMiam",
+      yesNoMiamPolicyUpgrade: true,
+      miamSelection: "attended4MonthsPrior",
+      subJourney: true,
+    });
+  });
+  test(`Complete the C100 Create case MIAM as a solicitor with the following options:
```
Comment (JakeCG) [inline e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts:44]: Needs a gap

File: e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+  Not Accessibility testing,
+  Not Error message testing,
+  Saying yes to yesExemption
+  miamSelection "attended4MonthsPrior"
+  Saying Yes to all options. @crossbrowserManageCases`, async ({
```
Comment (JakeCG) [inline e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts:49]: Remove the tag here

File: e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+      user: "solicitor",
+      accessibilityTest: false,
+      errorMessaging: false,
+      C100MiamPolicyUpgrade1PageType: "yesExemption",
+      yesNoMiamPolicyUpgrade: true,
+      miamSelection: "attended4MonthsPrior",
+      subJourney: true,
+    });
+  });
+  test(`Complete the C100 Create case MIAM as a solicitor with the following options:
```
Comment (JakeCG) [inline e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts:63]: Needs a gap

File: e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+  Not Accessibility testing,
+  Not Error message testing,
+  Saying yes to yesExemption
+  miamSelection "initiatedMIAMBeforeProceedings_MIAMCertificate"
+  Saying no to all options. @crossbrowserManageCases`, async ({
```
Comment (JakeCG) [inline e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts:68]: Remove the tag here

File: e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+      user: "solicitor",
+      accessibilityTest: false,
+      errorMessaging: false,
+      C100MiamPolicyUpgrade1PageType: "yesExemption",
+      yesNoMiamPolicyUpgrade: false,
+      miamSelection: "initiatedMIAMBeforeProceedings_MIAMCertificate",
+      subJourney: true,
+    });
+  });
+  test(`Complete the C100 Create case MIAM as a solicitor with the following options:
```
Comment (JakeCG) [inline e2e/tests/manageCases/createCase/C100/c100MiamPolicyUpgrade.spec.ts:82]: Needs a gap

**Review by JakeCG — CHANGES_REQUESTED** [review]

**Review by JakeCG — APPROVED** [review]

**Review by OLT2000 — APPROVED** [review]
lgtm

**Pull request merged**